packages/core/components/Button/Button.tsx

```diff
@@ -36,7 +36,7 @@ export const Button = ({
 
   useEffect(() => setLoading(loadingProp), [loadingProp]);
 
-  const ElementType = href ? "a" : onClick ? "button" : "div";
+  const ElementType = href ? "a" : "button";
```

Comment:
Ah OK, we can't break that.

FWIW relying on `onClick`is  not sufficent to ensure we always render a `button` when we need one, for example when using `type="submit"` inside a `form`. 

Fallback to `div` also creates a risk that we can fallback to an inaccessible pattern. I think it would be preferable to only render a `Button` as a `div` if this is explicity requested e.g. via a `dummyButton` or `buttonApperanceOnly` prop or similar.

I will cook something up 👍🏻

Reply:
Added a `routerLink` prop Button so we can render it as a`Link` 35240804454d5f575ea121cacdc00ed0cf360ed3

Reply:
Unfortunately we can't add a `routerLink` prop because it will bundle Next.js inside the core library.

Suggest we could either:

1. Add an extra check to the element condition to see if `type` is undefined (**my preference**)
2. Add an optional `el` prop or similar, with options `a`, `button`, `div`:

```tsx
<Link href="/docs" style={{ display: "flex" }}>
  <Button el="div">Read docs</Button>
</Link>
```

Reply:
Added extra condition on `type`.

Reply:
Perfect, it work